builtin/audit/file/backend_test.go

```diff
@@ -136,6 +136,45 @@ func TestAuditFile_fileMode0000(t *testing.T) {
 	}
 }
 
+func TestAuditFile_EventLogger_fileModeNew(t *testing.T) {
+	modeStr := "0777"
+	mode, err := strconv.ParseUint(modeStr, 8, 32)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	path, err := ioutil.TempDir("", "vault-test_audit_file-eventlogger-file_mode_new")
```

Comment:
In tests `t.TempDir()` is usually preferable these days.

Reply:
Ah ok. I will fix this test and the ones I stole this code from.

Reply:
You needn't fix existing tests, I'm just suggesting that for new tests this is the recommended method.

Reply:
I'll only fix the new test and the existing one that I copied. The other tests use ioutil.TempFile so I'll leave those alone for now.